packages/components/src/components/form/SegmentedButton/SegmentedButton.tsx

```diff
@@ -0,0 +1,87 @@
+import { Button as ChakraButton, IconProps, ResponsiveValue, forwardRef, useMultiStyleConfig } from '@chakra-ui/react';
+import { Flex } from '@components';
+import { useResponsiveValue } from '@hooks';
+import React, { FC, PropsWithChildren, ReactElement, cloneElement, isValidElement, useState } from 'react';
+import {
+  SegmentedButtonButtonProps,
+  SegmentedButtonProps,
+  SegmentedButtonSize,
+  SegmentedButtonStaticMembers,
+  SegmentedButtonStyles,
+  SegmentedIconButtonProps,
+  Value,
+} from './types';
+
+export const SegmentedButton: FC<PropsWithChildren<SegmentedButtonProps>> & SegmentedButtonStaticMembers = ({
+  children,
+  onChange,
+  variant = 'primary',
+  size = 'l',
+  defaultValue = '',
+  isDisabled = false,
+}) => {
+  const styles = useMultiStyleConfig('SegmentedButton') as SegmentedButtonStyles;
+  const [selectedValue, setSelectedValue] = useState<Value>(defaultValue);
+  const responsiveSize = useResponsiveValue(size) as SegmentedButtonSize;
+
+  const handleChange = (value: Value) => {
+    setSelectedValue(value);
+    onChange(value);
+  };
+
+  return (
+    <Flex {...styles.container} role="group">
+      {(children as ReactElement[]).map((child) =>
+        cloneElement(child, {
+          key: child.props.value,
+          onClick: () => handleChange(child.props.value),
+          isActive: selectedValue === child.props.value,
+          variant,
+          size: responsiveSize,
+          isDisabled,
+          'aria-current': selectedValue === child.props.value,
+          'data-value': child.props.value,
```

Comment:
@Claratj given that we are now using the `value` property I think we could omit the `data-value`. You'll need to update the tests too

Reply:
done!